### PR TITLE
Update libs for RocksJava Static build

### PR DESCRIPTION
**Pull request by adamretter**
Updates ZStd and Snappy to the latest versions.

### CODE REVIEW

**Review by mrambacher — APPROVED** [review]
LGTM.

**Comment by jay-zhuang** [comment]
Is the CI failure related to this change?

**Comment by adamretter** [comment]
@jay-zhuang Ooof! Yes, it looks like a problem with compiling Snappy on that specific platform, I will have to investigate:

```
/Users/distiller/project/snappy-1.1.9/snappy.cc:1033:36: error: invalid output constraint '=@ccz' in asm
      : [tag_type] "+r"(tag_type), "=@ccz"(is_literal));
                                   ^
1 error generated.
make[4]: *** [CMakeFiles/snappy.dir/snappy.cc.o] Error 1
make[3]: *** [CMakeFiles/snappy.dir/all] Error 2
make[2]: *** [all] Error 2
make[1]: *** [libsnappy.a] Error 2
```

**Review by ajkr — CHANGES_REQUESTED** [review]
The CI shows checksum mismatches. Do they need to be updated?

**Comment by adamretter** [comment]
@ajkr It is failing because downloading the artifacts yields HTTP 404 Not Found.

It looks like @jay-zhuang updated the CircleCI config overrides the URLs for the download of the artifacts.
@jay-zhuang Would you be able to upload the latest libraries please?

**Comment by jay-zhuang** [comment]
I think there's failure even build locally (without using binaries from s3), please take a look:
```
/home/zjay/ws/rocksdb/snappy-1.1.9/snappy.cc:1017:8: warning: always_inline function might not be inlinable [-Wattributes]
 1017 | size_t AdvanceToNextTag(const uint8_t** ip_p, size_t* tag) {
      |        ^~~~~~~~~~~~~~~~
/home/zjay/ws/rocksdb/snappy-1.1.9/snappy.cc: In function 'std::pair<const unsigned char*, long int> snappy::DecompressBranchless(const uint8_t*, const uint8_t*, ptrdiff_t, T, ptrdiff_t) [with T = char*]':
/home/zjay/ws/rocksdb/snappy-1.1.9/snappy.cc:1017:8: error: inlining failed in call to always_inline 'size_t snappy::AdvanceToNextTag(const uint8_t**, size_t*)': function body can be overwritten at link time
/home/zjay/ws/rocksdb/snappy-1.1.9/snappy.cc:1097:43: note: called from here
 1097 |         size_t tag_type = AdvanceToNextTag(&ip, &tag);
      |                           ~~~~~~~~~~~~~~~~^~~~~~~~~~~
```

zstd and sanppy binaries are uploaded.

**Comment by jowlyzhang** [comment]
I have uploaded the latest versions for these libraries. Can you rerun all the failing tests? Thanks!

File: Makefile [hunk 1]
```diff
@@ -2094,20 +2094,20 @@ ROCKSDB_JAVADOCS_JAR = rocksdbjni-$(ROCKSDB_JAVA_VERSION)-javadoc.jar
...
 SNAPPY_DOWNLOAD_BASE ?= https://github.com/google/snappy/archive
-LZ4_VER ?= 1.9.3
-LZ4_SHA256 ?= 030644df4611007ff7dc962d981f390361e6c97a34e5cbc393ddfbe019ffe2c1
+LZ4_VER ?= 1.9.4
+LZ4_SHA256 ?= 0b0e3aa07c8c063ddf40b082bdf7e37a1562bda40a0ff5272957f3e987e0e54b
 LZ4_DOWNLOAD_BASE ?= https://github.com/lz4/lz4/archive
-ZSTD_VER ?= 1.4.9
-ZSTD_SHA256 ?= acf714d98e3db7b876e5b540cbf6dee298f60eb3c0723104f6d3f065cd60d6a8
+ZSTD_VER ?= 1.5.5
+ZSTD_SHA256 ?= 98e9c3d949d1b924e28e01eccb7deed865eefebf25c2f21c702e5cd5b63b85e1
```
Comment (jowlyzhang) [inline Makefile:2110]: This one seems to have been updated to 9c4396cc829cfae319a6e2615202e82aad41372073482fce286fac78646d3ee4

**Comment by jowlyzhang** [comment]
Snappy still have some build errors.

**Comment by jowlyzhang** [comment]
Looks like snappy 1.1.9 has some compilation errors too.

**Comment by adamretter** [comment]
@cbi42 I am not seeing it also in the 8.7.fb branch yet?

**Comment by cbi42** [comment]
> @cbi42 I am not seeing it also in the 8.7.fb branch yet?

Yeah sorry, I just pushed it to 8.7.fb.

**Pull request closed without merging**